gms/feature_service.cc

```diff
@@ -51,19 +51,6 @@ feature_config feature_config_from_db_config(const db::config& cfg, std::set<sst
 
     fcfg._disabled_features = std::move(disabled);
 
-    switch (sstables::version_from_string(cfg.sstable_format())) {
-    case sstables::sstable_version_types::ka:
-    case sstables::sstable_version_types::la:
-    case sstables::sstable_version_types::mc:
-        fcfg._disabled_features.insert("MD_SSTABLE_FORMAT"s);
-        [[fallthrough]];
-    case sstables::sstable_version_types::md:
```

Comment:
What if it's configured to `md` in an existing scylla.yaml config file?

Reply:
@bhalevy i am reluctant to "lie" to user. because, if user explicitly specify `md` in `scylla.yaml` , we either follow it, or reject it. if we follow it, we fail to "make 'me' mandatory".  we can reject it though, in two ways. 

1. refuse to boot with an error message. this is what we have with this change.
2. boot anyway with or without an error message. this is very tolerant. but it still not what user expect. i'd say.

if we believe the 2nd option is better than the first one. i will implement it.

Reply:
okay, we can live with if we don't ignore silently but rather refuse to boot.

Pleas update also documentation:
- docs/architecture/sstable/index.rst
- docs/architecture/sstable/sstable3/index.rst

Reply:
ack. will go with "boot anyway with an error message". and will update the docs.